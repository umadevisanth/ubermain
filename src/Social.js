import React, { useState } from 'react';
import { View, Text, SafeAreaView, Image, TouchableOpacity, Linking } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import CheckBox from 'expo-checkbox';
import Globalstyle from './Globalstyle'; // Assuming Globalstyle contains global styles
import { Feather } from '@expo/vector-icons';

const Social = () => {
  // Social text content
  const social = {
    account: 'Choose an Account',
    facebook: 'Facebook',
    google: 'Google',
    click: 'By clicking on a social option you may receive an SMS verification. Message and data rates may apply.',
    next: 'Next',
    errorMessage: 'Please select a social option.', // New error message
  };

  const navigation = useNavigation();
  const [isChecked, setIsChecked] = useState(false);
  const [error, setError] = useState('');

  // Function to handle navigation when the "Next" button is pressed
  const handleNavigate = () => {
    if (isChecked) {
      navigation.navigate('First'); // Navigate to 'Phonenumber' screen only if checkbox is checked
    } else {
      setError(social.errorMessage); // Set error message if checkbox is not checked
    }
  };

  // Function to open Facebook URL
  const openFacebook = () => {
    Linking.openURL('https://www.facebook.com/');
  };

  // Function to open Gmail URL
  const openGmail = () => {
    Linking.openURL('https://mail.google.com/');
  };

  return (
    <SafeAreaView style={Globalstyle.socialsafeareaview}>
      {/* Navigation Back Button */}
      <View style={Globalstyle.socialnavigationbutton}>
        <TouchableOpacity onPress={() => navigation.goBack('PhoneNumber')}>
          <Feather name="arrow-left" size={24} color="#EDF6FF" style={Globalstyle.Social_iconleft} />
          {/* <i class="fa-solid fa-arrow-left"></i> */}
        </TouchableOpacity>
      </View>
      <View style={{ flex: 5 }}>
        {/* Account Selection */}
        <View style={Globalstyle.socialaccountselection}>
          <Text style={Globalstyle.Social_account}>{social.account}</Text>
        </View>

        {/* Social Options */}
        <View style={Globalstyle.socialsocialoptions}>
          <TouchableOpacity onPress={openFacebook}>
            <View style={Globalstyle.socialoptionview}>
              <Image style={Globalstyle.Social_facebookimage} source={require('../assets/facebook-logo-3.png')} />
              <Text style={Globalstyle.Social_facebook}>{social.facebook}</Text>
            </View>
          </TouchableOpacity>

          <TouchableOpacity onPress={openGmail}>
            <View style={Globalstyle.socialviewtext}>
              <Image style={Globalstyle.Social_googleimage} source={require('../assets/Google-Logo-PNG-Image.png')} />
              <Text style={Globalstyle.Social_google}>{social.google}</Text>
            </View>
          </TouchableOpacity>
        </View>

        {/* Checkbox and Information Text */}
        <View style={Globalstyle.socialcheckbox}>
          <CheckBox value={isChecked} onValueChange={setIsChecked} />
          <Text style={Globalstyle.socialchecktextmsg}>
            {social.click}
          </Text>
        </View>
      </View>

      {/* Error Message */}
      {error !== '' && <Text style={Globalstyle.socialerrormsg}>{error}</Text>}

      {/* Next Button */}
      <View style={{ flex: 1 }}>
        <TouchableOpacity
          style={{
            borderColor: isChecked ? '#EDF6FF' : 'grey', // Change border color based on checkbox state
            justifyContent: 'center',
            borderRadius: 2,
            width: '80%',
            marginLeft: '10%',
            opacity: isChecked ? 1 : 0.5, // Change opacity based on checkbox state
          }}
          onPress={handleNavigate}
          disabled={!isChecked} // Disable button if checkbox is not checked
        >
          <Text style={Globalstyle.socialdisablebutton}>{social.next}</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
};

export default Social;
