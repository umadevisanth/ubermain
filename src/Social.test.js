import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { useNavigation } from '@react-navigation/native';
import Social from '../src/Social';

// Mock @expo/vector-icons
jest.mock('@expo/vector-icons', () => ({
    AntDesign: ' AntDesign', // Mock the specific icon component you're using
}));

jest.mock('expo-checkbox', () => ({
  CheckBox: jest.fn(({ onValueChange }) => (
    <input type="checkbox" onChange={onValueChange} />
  )),
}));




// Mock useNavigation hook
jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
}));

describe('Social component', () => {
  it('renders correctly', () => {
    // Mock useNavigation behavior
    useNavigation.mockReturnValue({ navigate: jest.fn() });

    // Render the Login component
    const tree = renderer.create(<Social />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
