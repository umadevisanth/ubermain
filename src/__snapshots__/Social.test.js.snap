// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social component renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "black",
      "height": "100%",
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      < AntDesign
        color="black"
        name="left"
        size={20}
        style={
          {
            "color": "#979797",
            "height": 16,
            "left": 53,
            "top": 96,
            "width": 16,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "color": "#EDF6FF",
          "fontSize": 30,
          "height": 33,
          "left": 37,
          "top": 131,
          "width": 349,
        }
      }
    >
      Choose an Account
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../assets/facebook-logo-3.png",
          }
        }
        style={
          {
            "height": 30,
            "left": 53,
            "top": 196,
            "width": 30,
          }
        }
      />
      <Text
        style={
          {
            "color": "white",
            "fontSize": 24,
            "height": 33,
            "left": 127,
            "top": 196,
            "width": 409,
          }
        }
      >
        FaceBook
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View>
      <Image
        source={
          {
            "testUri": "../../../assets/Google-Logo-PNG-Image.png",
          }
        }
        style={
          {
            "height": 30,
            "left": 52,
            "top": 250,
            "width": 30,
          }
        }
      />
      <Text
        style={
          {
            "color": "white",
            "fontSize": 24,
            "height": 33,
            "left": 148,
            "top": 219,
            "width": 409,
          }
        }
      >
        Google
      </Text>
    </View>
  </View>
  <View>
    <Text
      style={
        {
          "color": "#EDF6FF",
          "fontSize": 14,
          "height": 33,
          "left": 37,
          "lineHeight": 16.41,
          "top": 302,
          "width": 409,
        }
      }
    >
      By clicking on a social option you may recieve an SMS for verification. Message and data rates may apply.
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "borderColor": "#EDF6FF",
        "borderRadius": 2,
        "borderWidth": 1,
        "height": 58,
        "justifyContent": "center",
        "left": 37,
        "opacity": 1,
        "top": 500,
        "width": 341,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 21,
          "left": 150,
        }
      }
    >
      Next
    </Text>
  </View>
</RCTSafeAreaView>
`;
